one-point-oh.md

```diff
@@ -0,0 +1,155 @@
+# 1.0 plan and priorities
+
+We don't have time for big changes or big features (dev time + time to marinate). We know vaguely where we're going (see other design PRs).
+
+High level priorities:
+
+- Stability and polish
+  - bugs, performance, glitches, etc.
+- User experience
+  - Goal: KCL is easy to read and comprehend
+  - Feedback is useful (though perhaps not pretty)
+    - IDE experience
+    - Error messages (parser in particular)
+- Supporting core new features in engine, as required
+- Future compatibility
+- Not making tech debt worse
+  - And ideally a little bit better
+
+Related:
+
+- [Modelling app roadmap](https://github.com/orgs/KittyCAD/projects/35)
+- [1.0 milestone](https://github.com/KittyCAD/modeling-app/milestone/2)
+- [CAD primitives roadmap](https://github.com/KittyCAD/modeling-app/issues/729)
+
+Priorities in **bold**, dependencies in *italics*, task size in t-shirt sizes (taking into account breaking change management; m-l means uncertainty about which, not 'somewhere in between').
+
+## As required
+
+### Bug fixing
+
+**P0**:
+
+- high priority bugs as they emerge
+- plan to seek out and address lower-priority bugs during feature freeze
+
+
+### Non-language feature work
+
+Priority follows from other team's priorities.
+
+New or more flexible std lib functions and code mods are expected as side effects of other work.
+
+New features which might require significant work:
+
+- [Multiple profiles in Sketches](https://github.com/KittyCAD/modeling-app/issues/1876)
+- Assemblies?
+
+
+### Pre-paying technical debt
+
+**P1**:
+
+- Don't make things worse by adding hacks, especially around the API of KCL
+
+**P3**:
+
+- Supplying user-side IDs to engine
+- code mod API
+
+
+## Language priorities
+
+**P1** (priority ordered):
+
+- reserve keywords (s)
+  - https://github.com/KittyCAD/modeling-app/issues/4486
+  - and give meaning to underscore in identifiers
+- function args
+  - named and optional args (m; *$0*)
+  - only one non-named arg (m; *$1*)
+    - ideally we make it the first arg, but doesn't have to be
+    - could require it be called `self`, but I don't think we need to and that makes back-compat harder
+- simple tidying changes
+  - automatic conversion of round floats to ints, remove the `int()` function (s)
```

Comment:
Sorry, I forgot to respond on Slack. My thinking is that although it is a pretty small thing, it is one of many small things that all together make KCL feel a bit rough, and while not needing `int` alone isn't much, if we can polish all or most of these rough edges, then KCL will feel a lot nicer, especially for new users.

There was also a lot of discussion about other numeric stuff (all of which I think we *should* postpone to after 1.0). To be clear, I think not requiring `int` is a really small amount of work and that it doesn't make things any harder for the other numeric stuff, so IMO the cost/benefit here is good.

Reply:
okay i just want to make sure we aren't creating magic, do other programming languages do this automatic conversion of floats to ints. MEs aren't so dense as to not know the difference and I think giving them the control to change the type is better than not, BUT i guess if we are only talking about changing 10.0000000000000000000... to a int then its fine, but if we are talking about changing 10.000000000000000001 i say no

Reply:
Could you say why you don't like the 10.000000000000000001 thing? We could definitely do just the former and it would get us most of the way there, but using a tiny epsilon would avoid some minor irritation for users. 

It's not so much that it's a hard concept, just not one which fits with the rest of the language - we just have a number type, we don't ever expose the int vs not-int distinction explicitly, and it leads to really awkward behaviour like `foo(5)` working but `x = 5; foo(5)` being an error which requires `x = 5; foo(int(5))`.

Reply:
Other languages do have an equivalence, e.g., Javascript has the position that all numbers are f64, but lets you use those in `for` loops, etc. (although their approach is to just accept floats everywhere rather than having an implicit conversion, I believe the conversion is better since `for i in [0..(0.1 + 0.2) * 10]` should either be an error or treat the upper bound as 3, whereas in JS the upper bound is equivalent to being treated like 4).

Example for why we might want a tiny epsilon is `(0.1 + 0.2) * 10` comes out at `3.0000000000000004` and I would love to treat that like `3`.

Reply:
I think with the precision we need for CAD I do not feel comfortable converting anything that is not 00000000 all zeroes without the user explicity saying to do so. Because for like a engine or a rocket they might actually want that.